src/librustc_back/target/windows_base.rs

```diff
@@ -63,7 +60,32 @@ pub fn opts() -> TargetOptions {
 
             // Always enable DEP (NX bit) when it is available
             "-Wl,--nxcompat".to_string(),
+
+            // Do not use the standard system startup files or libraries when linking
+            "-nostdlib".to_string(),
+        ),
+        pre_link_objects_exe: vec!(
+            "crt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        pre_link_objects_dll: vec!(
+            "dllcrt2.o".to_string(),
+            "rsbegin.o".to_string(),
+        ),
+        late_link_args: vec!(
+            "-lmingwex".to_string(),
+            "-lmingw32".to_string(),
+            "-lgcc".to_string(), // alas, mingw* libraries above depend on libgcc
+            "-lmsvcrt".to_string(),
+            "-ladvapi32".to_string(),
+            "-lshell32".to_string(),
+            "-luser32".to_string(),
+            "-lkernel32".to_string(),
```

Comment:
Ah, hm.  I stand corrected!

Well, on second look, there is a small chance of backcompat fall-out:
`nm -A -u libmingwex.a | grep __imp__` mostly shows symbols from `kernel32`, but not all of them are:
`lib32_libmingwex_a-getlogin.o:         U __imp__GetUserNameA@8` is from `advapi32.dll`,
`lib32_libmingwex_a-wassert.o:         U __imp__MessageBoxW@16` is from `user32.dll`

The last one actually breaks linking when building libflate, because I guess `miniz` uses `assert` internally.
There might be other crates who are in the same boat.